### PR TITLE
[Python][Callbacks][1/n] CallbackManager impl

### PR DESCRIPTION
**Pull request by LastMileNewGrad**
[Python][Callbacks][1/n] CallbackManager impl

For context, the callback typehint is defined as:
`Callback = Callable[[CallbackEvent], Awaitable]`

In laymans terms this means an async function that takes in a `CallbackEvent` (no output types are defined here).

Some python stuff going on here. When you define an async function, its really a function that returns a coroutine. a `coroutine` conforms to type `Awaitable`

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/119).
* #120
* #109
* #108
* #107
* __->__ #119

### CODE REVIEW

File: python/src/aiconfig/callback.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        return Ok(res)
+    except Exception as e:
+        return Err(str(e))
+
+
+class Record(BaseModel):
+    model_config = ConfigDict(strict=True, frozen=True)
+
+
+class pydanticBackedCallBackEvent(Record):
```
[the remark this answers is not in the document]
Comment (LastMileNewGrad) [reply python/src/aiconfig/callback.py:70]: It can be a pydantic class directly, reason I chose to do it this way is so that you can construct the class easily.

`CallbackEvent("event", {"input": data})`

where as with Pydantic, you have to do dictionary unpacking
`CallbackEvent(**{"event", {"input": data}})`

There isn't a need for the type validation Pydantic brings here, just the easy serialization & validation of objects.

File: python/src/aiconfig/callback.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
+# Standard Libraries
```
[the remark this answers is not in the document]
Comment (LastMileNewGrad) [reply python/src/aiconfig/callback.py:1]: Updated root imports

**Review by jonathanlastmileai — APPROVED** [review]

File: python/src/aiconfig/callback.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        # The name of the file that triggered the event.
+        self.file = file
+        # Anything available at the time the event happens.
+        # It is passed to the callback.
+        self.data = data
+        self.ts_ns = ts_ns
+
+
+# Type Aliases
+Callback = Callable[[CallbackEvent], Awaitable]
```
[the remark this answers is not in the document]
Comment (LastMileNewGrad) [reply python/src/aiconfig/callback.py]: updated to Any for now

File: python/src/aiconfig/callback.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        return CallbackManager([callback])
+
+
+# Default Callback Manager Logger to file
+def create_logging_callback(log_file: str = None) -> Callback:
+    """
+    Creates a callback that logs the event to a file.
+    """
+    if log_file is None:
+        log_file = "callbacks.log"
```
[the remark this answers is not in the document]
Comment (LastMileNewGrad) [reply python/src/aiconfig/callback.py:116]: added to gitignore
